### PR TITLE
Feature: AddressView with polygon overlay support

### PR DESCRIPTION
**Pull request by osanoj**
# Why?
A circle overlay is not perfect for showing postal area on a map. It would be much nicer to use a polygon to show the actual area.

# What?
Extended `AddressView` with possibility to show a polygon overlay instead of annotation or circle.

# Show me
Simulator Screen Shot - iPhone Xs - 2019-08-26 at 16 05 44
Simulator Screen Shot - iPhone Xs - 2019-08-26 at 16 05 50

### CODE REVIEW

**Comment by 3lvis** [comment]
😍

**Comment by osanoj** [comment]
Supporting multiple polygons:
Simulator Screen Shot - iPhone Xs - 2019-08-27 at 11 04 13

Also adjusted the fill color to make map underneath more visible (after designer feedback).

File: Sources/Fullscreen/AddressView/AddressView.swift [hunk 1]
```diff
@@ -136,14 +134,37 @@ public class AddressView: UIView {
...
+
+    public func makePolygonOverlayVisible() {
+        guard let firstPolygon = polygons.first else {
+            return
+        }
+        let mapRect = polygons.dropFirst().reduce(firstPolygon.boundingMapRect, { (mapRect, polygon) -> MKMapRect in
+            return polygon.boundingMapRect.union(mapRect)
+        })
+
+        let bottomInset = UIDevice.isIPad() ? 16 : 16 + .mediumLargeSpacing
```
Comment (vadymmarkov) [inline Sources/Fullscreen/AddressView/AddressView.swift]: Isn't 16 = `.mediumLargeSpacing`?

File: Demo/Fullscreen/AddressView/AddressViewDemoView.swift [hunk 2]
```diff
@@ -54,6 +58,33 @@ public class AddressViewDemoView: UIView, Tweakable {
...
+        let polygonDemoData = [
+            ["0010", "OSLO", [coordinatesFor0010]],
+            ["Multi polygons", "OSLO", [coordinatesFor0010, [[10.74, 59.92], [10.75, 59.92], [10.74, 59.91]]]]
+        ]
+        polygonDemoData.forEach({
+            let postalCode = ($0[0] as? String) ?? ""
+            let postalAreaName = ($0[1] as? String) ?? ""
+            let areas = ($0[2] as? [[[Double]]]) ?? [[[Double]]]()
```
Comment (vadymmarkov) [inline Demo/Fullscreen/AddressView/AddressViewDemoView.swift]: Is it possible to make it more readable? 😅

Comment (osanoj) [reply Demo/Fullscreen/AddressView/AddressViewDemoView.swift]: What is difficult about that line 😜

Comment (osanoj) [reply Demo/Fullscreen/AddressView/AddressViewDemoView.swift]: I made an update that I think makes this a bit better 😄

**Pull request merged**